### PR TITLE
[IMP] base: allow to create External ID from the record + tz

### PR DESCRIPTION
**Pull request by JKE-be**
This PR has 3 purposes:
- Allow to create xmlid from record
- Allow to duplicate an xmlid
- Display datetime with user tz instead of utc

Not in this PR
- Be sure to have the first xmlid and not the last
    -> seems already the case, I cannot reproduce locally and don't find example anymore on prod.

Task-internal-process

### CODE REVIEW

**Comment by JKE-be** [comment]
@robodoo rebase-ff

File: addons/web/static/src/legacy/debug_manager.js [hunk 1]
```diff
@@ -21,6 +23,25 @@ class GetMetadataDialog extends Component {
         await this.getMetadata();
     }
 
+    async onClickCreateXmlid() {
+        const context = _.extend({}, this.context, {
+            'default_module': '__custom__',
+            'default_res_id': this.state.id,
+            'default_model': this.props.res_model,
+        });
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/debug_manager.js:31]: we don't use `_` in new code (+ no need to quote object keys in JS)
```suggestion
        const context = Object.assign({}, this.context, {
            default_module: '__custom__',
            default_res_id: this.state.id,
            default_model: this.props.res_model,
        });
```

File: addons/web/static/src/legacy/debug_manager.js [hunk 2]
```diff
@@ -21,6 +23,25 @@ class GetMetadataDialog extends Component {
...
+        const adapterParent = standaloneAdapter({ Component });
+        const dialog = new FormViewDialog(adapterParent, {
+            context: context,
+            on_saved: () => this.getMetadata(),
+            disable_multiple_selection: true,
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/debug_manager.js:36]: I think this option has no impact on FormViewDialogs

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**